astropy/utils/masked/function_helpers.py

```diff
@@ -494,12 +500,12 @@ def copyto(dst, src, casting="same_kind", where=True):
     if not isinstance(dst, Masked) or isinstance(where, Masked):
         raise NotImplementedError
 
-    src_data, src_mask = get_data_and_mask(src)
-
-    if src_data is not None:
+    if src is np.ma.masked or src is np.ma.nomask:
+        src_mask = src is np.ma.masked
+    else:
+        src_data, src_mask = get_data_and_mask(src)
```

Comment:
Is it worth folding the numpy masked stuff into this function?
Or a utility function. There's a bit of repetition across the functions to make the same check.

Reply:
I thought about that too, and the original `Masked._get_data_and_mask` had an option just to do that - the problem is that for most uses outside of setting elements, one does not want `np.ma.masked` and `np.ma.nomask` to be recognized as special.

In the end, I felt that given that it saves only two lines (or 8 lines total in this module), it was not quite worth making a special routine. My preference would be to keep things as they are, but if you think it is better, I can make a special routine in the module; then, we can see if it turns out to be handy elsewhere too.

Reply:
Are there more setter methods which are not yet supported?

Reply:
Somewhat annoyingly, every `__setitem__` has to do a version of this dance, to recognize these special values. Eg., `Masked.__setitem__` and `Time.__setitem__`: https://github.com/astropy/astropy/blob/e5fa5cb4eb48050ba49ead981a19fd07da3e8619/astropy/utils/masked/core.py#L352-L358

https://github.com/astropy/astropy/blob/e5fa5cb4eb48050ba49ead981a19fd07da3e8619/astropy/time/core.py#L1200-L1207

But also in the new representations, frames, and `SkyCoord` work (#16845, #17016).

Though most of them are subtly different...

Anyway, perhaps best not to overthink this too much. One can refactor later.

Reply:
> EDIT: I see that the `Time` case does not deal with `np.ma.nomask`. Sigh.

Turns out this at least is wrong - `Time` does deal properly with it. Don't know how I missed that before.